server/src/main/java/org/elasticsearch/cluster/ClusterChangedEvent.java

```diff
@@ -242,11 +243,16 @@ public boolean nodesChanged() {
         return nodesRemoved() || nodesAdded();
     }
 
-    /**
-     * Returns the {@link ProjectsDelta} between the previous cluster state and the new cluster state.
-     */
-    public ProjectsDelta projectDelta() {
-        return projectsDelta;
+    public Set<ProjectId> addedProjects() {
+        return projectsDelta.added();
+    }
+
+    public Set<ProjectId> removedProjects() {
+        return projectsDelta.removed();
+    }
+
+    public Set<ProjectId> commonProjects() {
+        return projectsDelta.common(state);
```

Comment:
does this fit with the existing pattern of deltas? could you just use a set operation where ever this is needed (currentProjects - added)? Seems a bit odd to have this here. Maybe as a static method on two given metadata?

Reply:
Yeah I was not super happy with the changes either. I was trying to achieve (1) compute the common set once and (2) not storing the common set separately if it is the same as all projects. I reverted the changes and let the caller compute the common set of projects ca0ea7a10cb676deccd184059427a66140d1a2b8